### PR TITLE
Javadocs transaction2

### PR DESCRIPTION
**Pull request by siyopao**
https://scalar-labs.atlassian.net/browse/DLT-205

2/3 for transaction.

### CODE REVIEW

File: src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** */
+/** Perform CRUD operations on {@link DistributedStorage} */
```
Comment (yito88) [inline src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java]: It looks some operations don't operate on DistributedStorage.

File: src/main/java/com/scalar/database/transaction/consensuscommit/PrepareMutationComposer.java [hunk 2]
```diff
@@ -21,9 +21,11 @@
 import java.util.Optional;
 import javax.annotation.concurrent.NotThreadSafe;
 
+/** Prepare mutations in order to commit to {@link com.scalar.database.api.DistributedStorage}. */
```
Comment (yito88) [inline src/main/java/com/scalar/database/transaction/consensuscommit/PrepareMutationComposer.java]: It is not wrong. I think it is confusing a little.
`PrepareMutation` in this class means the mutation mutates values, metadata and changes the state of record to the `PREPARED` state.

How about this?
```
Compose mutations to prepare records for a transaction
```

File: src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java (hunk 1 above)
Comment (feeblefakie) [reply src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java]: I think it's misleading and not very informative. It should explain how it operates with snapshot and the storage.
Maybe, understanding snapshot isolation properly before writing a doc is good. 
Also, it's a comment for a class, please start with a noun.

File: src/main/java/com/scalar/database/transaction/consensuscommit/PrepareMutationComposer.java (hunk 2 above)
Comment (siyopao) [reply src/main/java/com/scalar/database/transaction/consensuscommit/PrepareMutationComposer.java]: Thank you! I used your suggestion.

File: src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java (hunk 1 above)
Comment (siyopao) [reply src/main/java/com/scalar/database/transaction/consensuscommit/CrudHandler.java]: Thank you for the comments. I updated based on your feedback, so please check it when you have time.

**Pull request closed without merging**